runelite-client/src/main/java/net/runelite/client/plugins/config/PluginListItem.java

```diff
@@ -147,32 +147,37 @@
 				String pluginId = pluginInfo.get("id");
 
 				hotSwapButton.setIcon(REFRESH_ICON);
-				externalPluginManager.uninstall(pluginId);
```

Comment:
I removed this because it seemed like a bug. Previously `uninstall` was invoked twice, once on the UI thread (here), and once in the `SwingWorker#doInBackground`. So I changed this to just run once in the background, then display the dialog on the foreground, and finally reload it in the background again.